src/clj/rems/workflow/dynamic.clj

```diff
@@ -615,104 +544,66 @@
-             (handler? application (:actor cmd)))
-      ;; when actor is both applicant and handler, combine the allowed states
-      (or (applicant-error application cmd)
-          (actor-is-not-handler-error application cmd)
-          (state-error application ::draft #_::withdrawn ::returned ::submitted ::approved)
-          success-result)
-      (if (= (:actor cmd) (:applicantuserid application))
-        (or (applicant-error application cmd)
-            (state-error application ::draft #_::withdrawn ::returned)
-            success-result)
-        (or (actor-is-not-handler-error application cmd)
-            (state-error application ::submitted ::approved)
-            success-result)))))
+  [cmd _application _injections]
+  (ok {:event/type :application.event/member-invited
+       :application/member (:member cmd)}))
 
 (defmethod command-handler ::remove-member
-  [cmd application injections]
-  (or (actor-is-not-handler-or-applicant-error application cmd)
-      (when (= (:applicantuserid application) (:userid (:member cmd)))
+  [cmd application _injections]
+  (or (when (= (:applicantuserid application) (:userid (:member cmd)))
         {:errors [{:type :cannot-remove-applicant}]})
       (when-not (contains? (set (map :userid (:members application)))
                            (:userid (:member cmd)))
         {:errors [{:type :user-not-member :user (:member cmd)}]})
-      {:success true
-       :result {:event/type :application.event/member-removed
-                :event/time (:time cmd)
-                :event/actor (:actor cmd)
-                :application/member (:member cmd)
-                :application/id (:application-id cmd)
-                :application/comment (:comment cmd)}}))
+      (ok {:event/type :application.event/member-removed
+           :application/member (:member cmd)
+           :application/comment (:comment cmd)})))
 
 (defmethod command-handler ::uninvite-member
-  [cmd application injections]
-  (or (actor-is-not-handler-or-applicant-error application cmd)
-      (when-not (contains? (set (map (juxt :name :email) (:invited-members application)))
+  [cmd application _injections]
+  (or (when-not (contains? (set (map (juxt :name :email) (:invited-members application)))
                            [(:name (:member cmd))
                             (:email (:member cmd))])
         {:errors [{:type :user-not-member :user (:member cmd)}]})
-      {:success true
-       :result {:event/type :application.event/member-uninvited
-                :event/time (:time cmd)
-                :event/actor (:actor cmd)
-                :application/member (:member cmd)
-                :application/id (:application-id cmd)
-                :application/comment (:comment cmd)}}))
+      (ok {:event/type :application.event/member-uninvited
+           :application/member (:member cmd)
+           :application/comment (:comment cmd)})))
+
+(defn- enrich-event [event cmd]
```

Comment:
I think it's better to name this more explicitly like `command->event-base` or `add-common-event-fields-from-command` style.